tests/unit/common/test_generate_table.py

```diff
@@ -159,18 +185,48 @@ def test_dynamic_table_region(self):
         test_table.add_row(my_col=3.0, indexed_col=[1.0, 3.0], optional_col2=.5)
         test_table.add_row(my_col=4.0, indexed_col=[2.0, 4.0], optional_col2=.5)
 
-        test_dtr_table = self.TestDTRTable(name='test_dtr_table', description='my table')
+        test_dtr_table = self.TestDTRTable(name='test_dtr_table', description='my table',
+                                           target_tables={'ref_col': test_table,
+                                                          'indexed_ref_col': test_table})
+        self.assertIs(test_dtr_table['ref_col'].table, test_table)
+        self.assertIs(test_dtr_table['indexed_ref_col'].target.table, test_table)
 
         test_dtr_table.add_row(ref_col=0, indexed_ref_col=[0, 1])
         test_dtr_table.add_row(ref_col=0, indexed_ref_col=[0, 1])
 
-        # DTR table attribute needs to be set manually
-        test_dtr_table['ref_col'].table = test_table
-        test_dtr_table['indexed_ref_col'].target.table = test_table
-
         np.testing.assert_array_equal(test_dtr_table['indexed_ref_col'].target.data, [0, 1, 0, 1])
         np.testing.assert_array_equal(test_dtr_table['ref_col'].data, [0, 0])
 
+    def test_dynamic_table_region_optional(self):
+        test_table = self.TestTable(name='test_table', description='my test table')
+        test_table.add_row(my_col=3.0, indexed_col=[1.0, 3.0], optional_col2=.5)
+        test_table.add_row(my_col=4.0, indexed_col=[2.0, 4.0], optional_col2=.5)
+
+        test_dtr_table = self.TestDTRTable(name='test_dtr_table', description='my table',
+                                           target_tables={'optional_ref_col': test_table,
+                                                          'optional_indexed_ref_col': test_table})
+        self.assertIs(test_dtr_table['optional_ref_col'].table, test_table)
```

Comment:
@rly Could you try adding a row to `test_dtr_table`? I think it will fail because we need to make sure that unindexed `DynamicTableRegion` dtypes get mapped to `int` dtypes for the corresponding docval arg

Reply:
oh, you did, on line 191 and 192. OK, yeah this looks great!

Reply:
I added another test there just in case